overlays/build-tools-in-build-inputs.nix

```diff
@@ -9,17 +9,111 @@ let
   inherit (import ../lib { inherit lib; }) checkMkDerivationFor;
 
   buildTools = [
+    "antlr"
+    "asciidoc"
+    "asciidoctor"
+    "autoconf"
+    "autogen"
+    "automake"
+    "automoc4"
+    "autoreconfHook"
+    "bazel"
+    "bc"
+    "bdf2psf"
+    "bison"
+    "bmake"
+    "breakpointHook"
+    "bzip2"
+    "bundler"
+    "darwin.cctools"
     "cmake"
+    "docutils"
+    "dos2unix"
+    "desktop-file-utils"
+    "docbook2x"
+    "docbook_xml_dtd_412"
+    "docbook_xml_dtd_42"
+    "docbook_xml_dtd_43"
+    "docbook_xml_dtd_44"
+    "docbook_xml_dtd_45"
+    "docbook-xsl-nons"
+    "docbook-xsl-ns"
+    "ensureNewerSourcesForZipFilesHook"
+    "extra-cmake-modules"
+    "flex"
+    "gawk"
+    "gcc"
+    "gcc10"
+    "getopt"
+    "git"
+    "gnustep-make"
+    "gnum4"
+    "gnumake"
+    "gnused"
+    "gogUnpackHook"
+    "go-md2man"
+    "gtk-doc"
+    "gzip"
+    "help2man"
+    "intltool"
+    "libtool"
+    "lit"
+    "lld"
+    "llvm"
+    "llvmPackages"
+    "makeSetupHook"
+    "makeWrapper"
+    "man"
+    "maven"
     "meson"
     "ninja"
+    "nodePackages.node-gyp"
+    "nodePackages.node-gyp-build"
+    "nodePackages.node-pre-gyp"
+    "pandoc"
+    "patch"
+    "patchelf"
     "pkg-config"
+    "poetry"
+    "pythonPackages.pytest"
```

Comment:
This will only match python2’s pytest. Should we add at least `python3.pkgs.pytest` or switch to some looser methods of detection than strict attribute equality?

Reply:
For my usecase python3Packages.pytest is enough but we should probably make it looser. Can we just use pytest?

Reply:
Currently, it compares the inputs with each of listed attributes so `pytest` would not work (there is no such attribute). We would have to switch to a different testing method.

Reply:
Not going to rework that right now. Removed pytest.